src/battle/partner/sushie.c

```diff
@@ -1,19 +1,153 @@
+
+        return ApiStatus_BLOCK;
+    }
+
+    partnerActor->scale.x = 1.0f;
+    partnerActor->scale.y = 1.0f;
+    partnerActor->scale.z = 1.0f;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802382A4_707F44(ScriptInstance* script, s32 isInitialCall) {
+    f32 var1 = script->varTable[0];
+    f32 var2 = script->varTable[1];
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_802382A4_707F44);
+    add_vec2D_polar(&var1, &var2, 300, atan2(var1, var2, script->varTable[3], script->varTable[4]));
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_80238328_707FC8);
+    script->varTable[3] = var1;
+    script->varTable[4] = var2;
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_80238388_708028);
+    return ApiStatus_DONE2;
+}
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_80238480_708120);
+ApiStatus func_80238328_707FC8(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    s32 var = script->varTable[0];
+
+    if (var > 0) {
+        script->varTable[0] = 1;
+    } else {
+        script->varTable[0] = 0;
+    }
+
+    script->varTable[15] = 0;
+    script->varTable[10] = var;
+    if (battleStatus->waterBlockTurnsLeft < var) {
+        battleStatus->waterBlockTurnsLeft = var;
+        battleStatus->waterBlockAmount = 1;
+        battleStatus->unk_43C->unk_0C->unk_10 = battleStatus->waterBlockTurnsLeft;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80238388_708028(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partnerActor = battleStatus->partnerActor;
+    Actor* playerActor = battleStatus->playerActor;
+
+    D_8023C070 = playFX_59(1, partnerActor->currentPos.x - 5.5, partnerActor->currentPos.y + 15.5, partnerActor->currentPos.z + 5, playerActor->currentPos.x, playerActor->currentPos.y, playerActor->currentPos.z, rand_int(10) * D_8023C060_70BD00 + 1, 30);
+    
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80238480_708120(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Bytecode* args = script->ptrReadPos;
+    f32 var1 = get_float_variable(script, *args++);
+    f32 var2 = get_float_variable(script, *args++);
+    f32 var3 = get_float_variable(script, *args++);
+    s32 effect = battleStatus->unk_A0;
+
+    if (effect) {
+        remove_effect(effect);
+    }
+
+    battleStatus->unk_A0 = playFX_5A(0, var1, var2, var3, 1.5f, 0);
+
+    return ApiStatus_DONE2;
+}
 
 INCLUDE_ASM(s32, "battle/partner/sushie", func_80238540_7081E0);
 
-INCLUDE_ASM(s32, "battle/partner/sushie", func_80238E30_708AD0);
+ApiStatus func_80238E30_708AD0(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* partnerActor = battleStatus->partnerActor;
+    f32 var = ((((f64) script->varTable[0] * 3) / 100) + 1);
+    f32 xScale = partnerActor->scale.x;
+    f32 yScale = partnerActor->scale.y;
+    f32 zScale = partnerActor->scale.z;
```

Comment:
Are these temps required or can you just do `partnerActor->scale.x += (var - partnerActor->scale.x) / 3;`?

Reply:
I tried that before, but it didn't match, so yeah